froide/document/views.py

```diff
@@ -148,3 +151,37 @@ def upload_documents(request):
         }
     )
     return render(request, "document/upload.html", {"form": form, "config": config})
+
+
+def allow_write_document(func):
+    @wraps(func)
+    def inner(request, pk, slug, *args, **kwargs):
+        document = get_object_or_404(Document, pk=pk, slug=slug)
+        if document.can_write(request):
+            return func(request, document, *args, **kwargs)
+        return render_403(request)
+
+    return inner
+
+
+@require_POST
+@allow_write_document
+def set_title(request, document):
+    return set_property(document, request, "title")
+
+
+@require_POST
+@allow_write_document
+def set_description(request, document):
+    return set_property(document, request, "description")
+
+
+def set_property(document, request, name):
+    value = request.POST.get(name, None)
+    if value is None:
+        return render_400(request)
+    setattr(document, name, value)
+    document.save()
```

Comment:
Here you could use [`update_fields` keyword argument](https://docs.djangoproject.com/en/4.0/ref/models/instances/#ref-models-update-fields) because you are really only dealing with one field.

Reply:
Nice, I didn't know about this option